tensorzero-core/src/inference/types/file.rs

```diff
@@ -100,34 +172,107 @@ pub fn serialize_with_file_data<T: Serialize>(value: &T) -> Result<Value, Error>
     })
 }
 
+/// A file that can be located at a URL
+#[derive(Clone, Debug, PartialEq, Serialize, ts_rs::TS)]
+#[ts(export)]
+pub struct UrlFile {
+    pub url: Url,
+    #[ts(type = "string | null")]
+    pub mime_type: Option<MediaType>,
+}
+
+/// A file stored in an object storage backend, without data.
+/// This struct can be stored in the database. It's used by `StoredFile` (`StoredInput`).
+/// Note: `File` supports both `ObjectStorageFile` and `ResolvedObjectStorageFile`.
+#[derive(Clone, Debug, PartialEq, Serialize, ts_rs::TS)]
+#[ts(export)]
+pub struct ObjectStorageFile {
```

Comment:
can we actually call it UnresolvedObjectStorageFile? completely unambiguous when to use then